src/Sylius/Component/Core/Taxation/Processor/OrderTaxesProcessor.php

```diff
@@ -86,18 +86,20 @@ public function process(OrderInterface $order)
     }
 
     /**
-     * @param AddressInterface|null $shippingAddress
+     * @param OrderInterface $order
      *
      * @return ZoneInterface|null
      */
-    private function getTaxZone(AddressInterface $shippingAddress = null)
+    private function getTaxZone(OrderInterface $order)
     {
+        $shippingAddress = $order->getShippingAddress();
         $zone = null;
+
         if (null !== $shippingAddress) {
             $zone = $this->zoneMatcher->match($shippingAddress);
```

Comment:
Totally agreed 👍 However this PR is only about moving *default tax zone* and *default tax calculation strategy* to channel, so such a refactoring could and should be done in separate PR ;)